### PR TITLE
Fix answering calls (with pidgin-skype)

### PR DESCRIPTION
**Pull request by cwendling**
libskype: Don't use freed memory.  Don't expect the passed-in call number string to stay around, so duplicate it if needed.

Also fix a memory leak when media creation failed in `skype_media_initiate()`.

### CODE REVIEW

File: libskype.c [hunk 1]
```diff
@@ -3444,11 +3443,13 @@ skype_media_initiate(PurpleAccount *account, const char *who, PurpleMediaSession
...
 	}
-	callnumber_string = g_new(gchar, 10+1);
-	sscanf(temp, "CALL %s ", callnumber_string);
-	
 	if (media != NULL)
 	{
+		gchar *callnumber_string;
+
+		callnumber_string = g_new(gchar, 10+1);
+		sscanf(temp, "CALL %s ", callnumber_string);
```
Comment (cwendling) [inline libskype.c]: That's the leak fix: if `media` is `NULL` the call string isn't needed.

File: libskype.c [hunk 2]
```diff
@@ -3491,25 +3492,27 @@ skype_handle_incoming_call(PurpleConnection *gc, char *callnumber_string)
...
 	
-	who = g_strdup(&temp[21+strlen(callnumber_string)]);
+	who = g_strdup(&temp[21+strlen(callnumber_str)]);
 	g_free(temp);
 	
 	media = purple_media_manager_create_media(purple_media_manager_get(), purple_connection_get_account(gc), "fsrtpconference", who, FALSE);
 	
 	if (media != NULL)
 	{
+		gchar *callnumber_string = g_strdup(callnumber_str);
```
Comment (cwendling) [inline libskype.c]: That's the real fix: the `callnumber_str` argument comes from `skype_handle_received_message()`, which will free it when returning, but we need to keep it around (in the hash table, among other things).

**Comment by cwendling** [comment]
If you prefer I can provide a single-line patch just to fix the real issue, and leave the other changes out.

**Comment by EionRobb** [comment]
The old old old deprecated legacy defunct plugin still works?  Crazy :)

**Pull request merged**